staging/src/k8s.io/controller-manager/config/types.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LeaderMigrationConfiguration provides versioned configuration for all migrating leader locks.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type LeaderMigrationConfiguration struct {
+	metav1.TypeMeta
+
+	// LeaderName is the name of the leader election resource that protects the migration
+	// E.g. 1-20-KCM-to-1-21-CCM
+	LeaderName string
+
+	// ResourceLock indicates the resource object type that will be used to lock
+	// Should be "leases" or "endpoints"
+	ResourceLock string
```

Comment:
Wish we had made this an enum, oh well.

Reply:
Let's make room for future resource lock types, considering https://github.com/kubernetes/kubernetes/blob/8deb70c2ad6df1c02f88c0eec521c8ca1a4829f7/staging/src/k8s.io/component-base/config/v1alpha1/types.go#L50 They did not foresee `leases` either.

Reply:
Why not go ahead and make this an enum? (i.e., give it its own type?)

I guess that is backwards compatible and can be added in the future.

Reply:
I'd like to be consistent with other instances of resource type. I think we should but that seems like clean up better done in a separate PR.